service/controller/clusterapi/v29/key/common.go

```diff
@@ -6,6 +6,10 @@ import (
 	"github.com/giantswarm/aws-operator/pkg/label"
 )
 
+func ClusterID(getter LabelsGetter) string {
+	return getter.GetLabels()[label.Cluster]
```

Comment:
This makes me thing if it isn't better use pointer for `cr`. This would also make functions like `ToAWSConfig` nicer because our defaulting to zero value is weird.

Reply:
I had the same thoughts but I am not sure about it just yet. Let's keep it in mind and see how it goes.